packages/proto-loader/src/index.ts

```diff
@@ -115,6 +115,39 @@ export interface EnumTypeDefinition extends ProtobufTypeDefinition {
   format: 'Protocol Buffer 3 EnumDescriptorProto';
 }
 
+export enum IdempotencyLevel {
+  IDEMPOTENCY_UNKNOWN = 0,
+  NO_SIDE_EFFECTS = 1,
+  IDEMPOTENT = 2
+}
+
+export interface NamePart {
+  name_part: string;
+  is_extension: boolean;
+}
+
+export interface UninterpretedOption {
+  name?: NamePart[];
+  identifier_value?: string;
+  positive_int_value?: number;
+  negative_int_value?: number;
+  double_value?: number;
+  string_value?: string;
+  aggregate_value?: string;
+}
+
+export interface CustomHttpPattern {
+  kind?: string;
+  path?: string;
+}
+
+export interface MethodOptions {
+  deprecated?: (boolean|null);
+  idempotency_level?: (IdempotencyLevel|keyof typeof IdempotencyLevel|null);
+  uninterpreted_option?: (UninterpretedOption[]|null);
```

Comment:
I don't like having these fields as optional and/or nullable. `deprecated` can coalesce to `false`, `idempotency_level` can coalesce to the default value, and `uninterpreted_option` can coalesce to `[]`. Similarly, `idempotency_level` can be narrowed to either `IdempotencyLevel` or `keyof typeof IdempotencyLevel`. In all of these cases, if the value provided by protobuf.js has a wider type, it should be explicitly narrowed in code in this library.

Reply:
I cleaned up these nullable values, and add a new test to make sure that protobuf.js won't have any fallback.

Reply:
These fields should also not be optional, and `idempotency_level` should be exactly one of `IdempotencyLevel` and `keyof typeof IdempotencyLevel`. My goal here is for values provided by this library to the user to have the narrowest types possible, and we can make these narrower because we control the code that returns them. We can explicitly set a value for each field that is not already set. We can convert `idempotency_level` to the chosen type if it has the wrong type.